### PR TITLE
enhance: setCurrentBrowserPathname() takes arguments

### PR DESCRIPTION
**Pull request by ntucker**
BREAKING CHANGE: Must send arguments to setCurrentBrowserPathname.
Also required initialPath argument to <BrowserPathname/>

### CODE REVIEW

File: packages/react-browser-pathname/src/index.tsx [hunk 1]
```diff
@@ -1,24 +1,42 @@
...
-    setLocation(getPathname());
-  }, []);
+  const setCurrentBrowserPathname = useCallback(
+    ({ data, title, url, type }: SET_ACTION) => {
+      switch (type) {
+        case 'REPLACE':
+          window.history.replaceState(data, title, url);
+          break;
+        case 'PUSH':
+          window.history.pushState(data, title, url);
```
Comment (0xcaff) [inline packages/react-browser-pathname/src/index.tsx:24]: title doesn't do anything. https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

File: packages/react-browser-pathname/src/index.tsx [hunk 2]
```diff
@@ -1,24 +1,42 @@
...
+  const [location, setLocation] = useState(initialPath);
 
-  const setCurrentBrowserPathname = useCallback(() => {
-    setLocation(getPathname());
-  }, []);
+  const setCurrentBrowserPathname = useCallback(
+    ({ data, title, url, type }: SET_ACTION) => {
+      switch (type) {
+        case 'REPLACE':
+          window.history.replaceState(data, title, url);
```
Comment (0xcaff) [inline packages/react-browser-pathname/src/index.tsx:21]: drop title

File: packages/react-browser-pathname/src/context.ts [hunk 3]
```diff
@@ -1,7 +1,14 @@
 import React, { useContext, useCallback, useEffect, useState } from 'react';
 import type { ReactNode } from 'react';
 
+export type SET_ACTION = {
+  url?: string;
```
Comment (0xcaff) [inline packages/react-browser-pathname/src/context.ts:5]: why is this allowed to be optional?

File: packages/react-browser-pathname/src/context.ts [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
+  title: string;
+  data: object;
```
Comment (0xcaff) [inline packages/react-browser-pathname/src/context.ts:7]: i'd avoid exposing data unless you plan to expose it in `onChange` and current location also

**Review by 0xcaff — APPROVED** [review]
pls clean up the interface but i think this is fine

File: packages/react-browser-pathname/src/index.tsx (hunk 1 above)
Comment (ntucker) [reply packages/react-browser-pathname/src/index.tsx:24]: lol this api is so dumb, why is the unused thing the second arg

File: packages/react-browser-pathname/src/context.ts (hunk 3 above)
Comment (ntucker) [reply packages/react-browser-pathname/src/context.ts:5]: https://developer.mozilla.org/en-US/docs/Web/API/History/pushState i dunno man

Comment (ntucker) [reply packages/react-browser-pathname/src/context.ts:5]: i think it might be for setting state without changing path

Comment (ntucker) [reply packages/react-browser-pathname/src/context.ts:5]: " If this parameter isn't specified, it's set to the document's current URL." yup - useful for state

**Pull request merged**